File: haohaninfo/回測範例/42-1.py
# -*- coding: UTF-8 -*-
#取I020，依照逗點分隔，並將分隔符號去除
I020 = [ line.strip('\n').split(",") for line in open('Futures_20170815_I020.csv')][1:]

#定義變數初始值
high=int(I020[0][4])
low=int(I020[0][4])

#開始計算高低點
for i in I020[1:]:
 price = int(i[4])
 if price > high:
  high=price
 if price < low:
  low=price
 print "Time:",i[0]," Price:",price," High:",high," Low:",low

File: haohaninfo/交易回測案例/42-1.R
# 讀取資料
I020 <- read.csv('Futures_20170815_I020.csv')
# 設定高低點初始值
highprice=0
lowprice=99999999
for (i in 1:nrow(I020)){
 if(highprice<I020[i,5]){highprice=I020[i,5]}
 if(lowprice>I020[i,5]){lowprice=I020[i,5]}
 cat(I020[i,1],"High Price:",highprice,"Low Price",lowprice,"\n")
}